extras/CatchAddTests.cmake

```diff
@@ -161,6 +161,22 @@ function(catch_discover_tests_impl)
     endif()
 
     list(APPEND tests "${prefix}${test}${suffix}")
+
+    string(JSON tags GET "${test_spec}" "tags")
+    string(JSON tags_length LENGTH "${tags}")
+
+    if("${tags_length}" GREATER 0)
+      math(EXPR tag_end "${tags_length} - 1")
+
+      foreach(tag_index RANGE "${tag_end}")
+        string(JSON tag GET "${tags}" "${tag_index}")
+        add_command(set_tests_properties
+          "${prefix}${test}${suffix}"
+          PROPERTIES
+            LABELS "${tags}"
+        )
+      endforeach()
```

Comment:
I am not sure why, but this does not do what it is supposed to. The resulting ctest script looks like this:

```
add_test( [==[And now a test case with weird tags.]==] /home/xarn/builds/Catch2/uyha-json-reporter/tests/ctest-registration-test/tests [==[And now a test case with weird tags.]==]  )
set_tests_properties( [==[And now a test case with weird tags.]==] PROPERTIES WORKING_DIRECTORY /home/xarn/builds/Catch2/uyha-json-reporter/tests/ctest-registration-test)
set_tests_properties( [==[And now a test case with weird tags.]==] PROPERTIES LABELS [==[[ "foo,bar", "tl;dr", "tl;dw" ]]==])
set_tests_properties( [==[And now a test case with weird tags.]==] PROPERTIES LABELS [==[[ "foo,bar", "tl;dr", "tl;dw" ]]==])
set_tests_properties( [==[And now a test case with weird tags.]==] PROPERTIES LABELS [==[[ "foo,bar", "tl;dr", "tl;dw" ]]==])
```

Which has the obvious issue of setting the same set of labels multiple times, but also it should set each tag as its own label, rather than the amalgamation of all tags as one big label.


Also check #1658 for proper handling of labels -> they need to be appended, rather than just set.

Reply:
The reason the amalgamation is there is `LABELS "${tags}"`, rather than `LABELS "${tag}"`.